gmso/external/convert_foyer.py

```diff
@@ -0,0 +1,611 @@
+from warnings import warn
+
+import numpy as np
+import unyt as u
+import os
+import sympy
+import pathlib
+
+from lxml import etree
+from gmso.utils.io import has_foyer  # Need for foyer
+from gmso.exceptions import ForceFieldParseError
+
+if has_foyer:
+    import foyer
+
+
+def from_foyer(foyer_xml, gmso_xml=None):
+    """Convert a foyer XML to a gmso XML
+
+    Parameters
+    ----------
+    foyer_xml : XML file
+        An XML file in the foyer format 
+   
+    Returns
+    -------
+    gmso_xml : XML file, default=None
+        An XML file in the gmso format
+    """
+
+    if gmso_xml is None:
+        stem_file_name = pathlib.Path(str(foyer_xml)).stem
+        suffix = "_gmso.xml"
+        gmso_xml = pathlib.Path(stem_file_name + suffix)
+    else:
+        gmso_xml = pathlib.Path(gmso_xml)
+
+    foyer_xml_tree = etree.parse(foyer_xml)
```

Comment:
We might want to convert `foyer_xml` into a pathlib `Path` object too. Although it does not seem that they support passing in a pathlib object yet. See [this issue/feature request](https://bugs.launchpad.net/lxml/+bug/1820837).

But that way, we would just have to remove a `str` call around the path object.

```python3
foyer_xml = pathlib.Path(str(foyer_xml))
foyer_xml_tree = etree.parse(str(foyer_xml))
```